src/selectors/cashRegister.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import { createSelector } from 'reselect';
+import { pageStateSelector } from './pageSelectors';
+
+import currency from '../localization/currency';
+
+/**
+ * Derives current balance from cash register transactions.
+ * @return {Number}
+ */
+export const selectTransactions = createSelector(
+  [pageStateSelector],
+  pageState => pageState.backingData
+);
+
+export const selectReceipts = createSelector([selectTransactions], transactions =>
+  transactions.filter(({ type }) => type === 'receipt')
```

Comment:
you can use `filtered` if your backing data is just all `Transaction`  - pretty sure it'd be faster regardless

Reply:
Unfortunately not possible, this is where the realm issue occurs 😢 .

Reply:
Update: whether realm bug or something else theoretical now as `transactions` is not longer guaranteed to be a realm collection (workaround for rerender during transition).